src/lib.rs

```diff
@@ -52,7 +52,7 @@ pub use crate::tree_state::TreeState;
 /// ```
 #[derive(Debug, Clone)]
 pub struct Tree<'a, Identifier> {
-    items: Vec<TreeItem<'a, Identifier>>,
+    items: &'a Vec<TreeItem<'a, Identifier>>,
```

Comment:
Just noticed this while glancing over this. When there is a reference, a slice would be better than a Vec.

```suggestion
    items: &'a [TreeItem<'a, Identifier>],
```